### ISSUE TITLE
False positive when running interfacer against gin.Engine

### ISSUE THREAD

**Issue by stephen-obashitech** [golangci/golangci-lint#554]
Thank you for creating the issue!

Please include the following information:
1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)
`golangci-lint has version v1.16.0 built from (unknown, mod sum: "h1:PcWAN9JHflZzJQaZVY1JXZE0Tgjq+jO2v4QLqJ/Azvw=") on (unknown)`

2. Config file: `cat .golangci.yml`
No config, but I used the following command
`golangci-lint run -v --disable-all -E interfacer`

3. Go environment: `go version && go env`
```
go version go1.12.4 linux/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/parallels/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/parallels/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/parallels/work/bugReports/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build053733424=/tmp/go-build -gno-record-gcc-switches"
```

4. Verbose output of running: `golangci-lint run -v`
```
golangci-lint run -v --disable-all -E interfacer
INFO [config_reader] Config search paths: [./ /home/parallels/work/bugReports /home/parallels/work /home/parallels /home /] 
INFO [lintersdb] Active 1 linters: [interfacer]   
INFO [loader] Go packages loading at mode load deps types and syntax took 2.241373592s 
INFO [loader] SSA repr building timing: packages building 47.388315ms, total 326.891576ms 
INFO [runner] worker.2 took 17.968µs              
INFO [runner] worker.1 took 18.538703ms with stages: interfacer: 18.529161ms 
INFO [runner] Workers idle times: #2: 10.634261ms 
INFO [runner] processing took 131.399µs with stages: exclude: 44.986µs, source_code: 23.578µs, identifier_marker: 13.873µs, cgo: 12.749µs, path_prettifier: 10.492µs, autogenerated_exclude: 9.524µs, nolint: 6.609µs, skip_dirs: 3.674µs, max_same_issues: 1.441µs, uniq_by_line: 1.123µs, max_from_linter: 1.041µs, path_shortener: 809ns, max_per_file_from_linter: 445ns, replacement_builder: 391ns, diff: 245ns, skip_files: 234ns, exclude-rules: 185ns 
main.go:20:21: `engine` can be `net/http.Handler` (interfacer)
func PerformRequest(engine *gin.Engine, method, url string, data []byte) *httptest.ResponseRecorder {
                    ^
INFO File cache stats: 1 entries of total size 749B 
INFO Memory: 24 samples, avg is 255.7MB, max is 538.4MB 
INFO Execution took 2.640685377s                  
Makefile:2: recipe for target 'all' failed
make: [all] Error 1 (ignored)
go run main.go
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:	export GIN_MODE=release
 - using code:	gin.SetMode(gin.ReleaseMode)

[GIN-debug] GET    /stuff                    --> main.stuffHandler (1 handlers)
```

The core of the issue is that interfacer suggests replacing the `gin.Engine` struct with the `net/http.Handler` interface in a function. However following that suggestion results in an error, since the `http.Handler` interface does not support the `ServeHTTP` method that `gin.Engine` does. I've attached the smallest code I can find which replicates the issue

```
package main

import (
	"bytes"
	"github.com/gin-gonic/gin"
	"net/http"
	"net/http/httptest"
)

func main() {

	TestEngine := gin.New()
	TestEngine.GET("/stuff", stuffHandler)

	PerformRequest(TestEngine, "POST", "http://localhost/stuff", []byte(""))
}

// PerformRequest will perform the given http request (from the method + url) using the given Gin Engine. It adds
// the provided data to the request body
func PerformRequest(engine *gin.Engine, method, url string, data []byte) *httptest.ResponseRecorder {
	recorder := httptest.NewRecorder()
	request := httptest.NewRequest(method, url, bytes.NewBuffer(data))
	engine.ServeHTTP(recorder, request)

	return recorder
}

func stuffHandler(c *gin.Context) {
	// stuff...
	c.Status(http.StatusOK)
}
```

**Comment by stephen-obashitech** [comment]
Closed because it turns out that it does contain the same method, I failed to notice I was using `*http.Handler` rather than `http.Handler`
